packages/marketplace/src/components/pages/developer-app-detail/developer-app-detail.tsx

```diff
@@ -0,0 +1,118 @@
+import { useHistory } from 'react-router'
+
+export type DeveloperAppDetailProps = {}
+
+export const handleOnDeleteAppSuccess = history => {
+  return () => {
+    history.replace(routes.DEVELOPER_MY_APPS)
+  }
+}
+
+export const renderAppHeaderButtonGroup = (
+  id,
+  appDetailState,
+  setIsAppRevisionComparisionModalOpen,
+  setIsDeleteModalOpen,
+  setIsInstallationsModalOpen,
+) => {
+  return (
+    <>
+      {id && (
+        <DeveloperAppDetailButtonGroup
+          appDetailState={appDetailState}
+          setIsAppRevisionComparisionModalOpen={setIsAppRevisionComparisionModalOpen}
+          setIsDeleteModalOpen={setIsDeleteModalOpen}
+          setIsInstallationsModalOpen={setIsInstallationsModalOpen}
+        />
+      )}
+    </>
+  )
+}
+
+export const closeInstallationsModal = (setIsInstallationsModalOpen: (isVisible: boolean) => void) => {
+  return () => {
+    setIsInstallationsModalOpen(false)
+  }
+}
+
+export const closeAppRevisionComparisionModal = (
+  setIsAppRevisionComparisionModalOpen: (isVisible: boolean) => void,
+) => {
+  return () => {
+    setIsAppRevisionComparisionModalOpen(false)
+  }
+}
+
+export const closeDeleteAppModal = (setIsDeleteModalOpen: (isVisible: boolean) => void) => {
+  return () => {
+    setIsDeleteModalOpen(false)
+  }
+}
+
+const DeveloperAppDetail: React.FC<DeveloperAppDetailProps> = () => {
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = React.useState(false)
+  const [isInstallationsModalOpen, setIsInstallationsModalOpen] = React.useState(false)
+  const [isAppRevisionComparisionModalOpen, setIsAppRevisionComparisionModalOpen] = React.useState(false)
+
+  const history = useHistory()
+  const appDetailState = useSelector(selectAppDetailState)
+  const appDetailData = useSelector(selectAppDetailData)
+  const isLoadingAppDetail = useSelector(selectAppDetailLoading)
+  const loginType = useSelector(selectLoginType)
+  const { id = '', name = '' } = appDetailData
+
+  return (
+    <div className={styles.appDetailContainer}>
+      <AppHeader
+        appDetailData={appDetailData}
+        buttonGroup={renderAppHeaderButtonGroup(
+          id,
+          appDetailState,
+          setIsAppRevisionComparisionModalOpen,
+          setIsDeleteModalOpen,
+          setIsInstallationsModalOpen,
+        )}
+      />
+      <AppContent appDetailData={appDetailData} loginType={loginType} />
+      <AppDelete
+        appId={id}
+        appName={name}
+        afterClose={closeDeleteAppModal(setIsDeleteModalOpen)}
+        visible={isDeleteModalOpen}
+        onDeleteSuccess={handleOnDeleteAppSuccess(history)}
+      />
+
+      <AppInstallations
+        appId={id}
+        appName={name}
+        visible={isInstallationsModalOpen}
+        afterClose={closeInstallationsModal(setIsInstallationsModalOpen)}
+        onUninstallSuccess={closeInstallationsModal(setIsInstallationsModalOpen)}
+      />
+
+      <AppRevisionModal
```

Comment:
I think should unmount and remove from DOM tree modal `isAppRevisionComparisionModalOpen && <AppRevisionModal ...` if we don't show it on UI